OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignalPrefs.java

```diff
@@ -204,6 +204,8 @@ else if (value instanceof Long)
                             editor.putLong(key, (Long)value);
                         else if (value instanceof Set)
                             editor.putStringSet(key, (Set<String>)value);
+                        else if (value == null)
+                            editor.remove(key);
```

Comment:
Or perhaps this should be `editor.putString(key, null)`? 
There is some difference with how `remove()` is treated as it is done when `commit()` is called, whereas the other put methods are written once `apply()` or `commit()` is called.

But the current changes is working.